lumibot/backtesting/backtesting_broker.py

```diff
@@ -484,21 +448,25 @@ def process_pending_orders(self, strategy):
                 # This is a hack to get around the fact that we need to get the previous day's data to prevent lookahead bias.
                 ohlc = strategy.get_historical_prices(
                     asset,
-                    1,
+                    2,
                     quote=order.quote,
-                    timeshift=-1,
+                    timeshift=-2,
                     timestep=self.data_source._timestep,
                 )
+                df = ohlc.df
+
+                # Make sure that we are only getting the prices for the current time exactly or in the future
+                df = df[df.index >= self.datetime]
 
                 if ohlc is None:
                     self.cancel_order(order)
                     continue
-                dt = ohlc.df.index[-1]
-                open = ohlc.df["open"].iloc[-1]
-                high = ohlc.df["high"].iloc[-1]
-                low = ohlc.df["low"].iloc[-1]
-                close = ohlc.df["close"].iloc[-1]
-                volume = ohlc.df["volume"].iloc[-1]
+                dt = df.index[0]
+                open = df["open"].iloc[0]
+                high = df["high"].iloc[0]
+                low = df["low"].iloc[0]
+                close = df["close"].iloc[0]
+                volume = df["volume"].iloc[0]
```

Comment:
Even though you are now pulling up to 2 candles, don't you still want the most recent (```iloc[-1]```) instead of 0?

- 2 candles returned:
  - .iloc[-1] --> give most recent, which is the correct price
  - .iloc[0] --> gives the first candle, which is now out of date
- 1 candle returned:
  - .iloc[-1] --> still gives the most recent, even though the timestamp is "2 candles ago"
  - .iloc[0] --> returns the same as .iloc[-1]